File: app/src/main/java/com/example/android_study/_view/_recyclerview/item_touch_helper/adapter/RvAdapter.java
package com.example.android_study._view._recyclerview.item_touch_helper.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.example.android_study.R;
import com.example.android_study._view._recyclerview.item_touch_helper.onMoveAndSwipedListener;
import com.example.android_study._view._recyclerview.nest_recycler.adapter.Datas;

import java.util.Collections;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Author: create by RhythmCoder
 * Date: 2020/5/22
 * Description:
 */
public class RvAdapter extends RecyclerView.Adapter<RvAdapter.Holder> implements onMoveAndSwipedListener {
    private LinearLayoutManager mLayoutMagager;
    List<Datas> mItems;
    Context mContext;

    public RvAdapter(Context context, List<Datas> mDatas) {
        this.mContext = context;
        this.mItems = mDatas;
    }

    public RvAdapter(LinearLayoutManager linearLayoutManager, Context context, List<Datas> mDatas) {
        this.mContext = context;
        this.mItems = mDatas;
        this.mLayoutMagager = linearLayoutManager;
    }

    @NonNull
    @Override
    public Holder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View inflate = LayoutInflater.from(mContext).inflate(R.layout.item_recycler_view, parent, false);
        return new Holder(inflate);
    }

    @Override
    public void onBindViewHolder(@NonNull Holder holder, int position) {
        holder.tvTitle.setText(mItems.get(position).title);
    }

    @Override
    public int getItemCount() {
        return mItems.size();
    }

    @Override
    public boolean onItemMove(int fromPosition, int toPosition) {
        Collections.swap(mItems, fromPosition, toPosition);
        notifyItemMoved(fromPosition, toPosition);
        return true;
    }

    @Override
    public void onItemDismiss(final int position) {
        mItems.remove(position);
        notifyItemRemoved(position);

//        Snackbar.make(parentView, context.getString(R.string.item_swipe_dismissed), Snackbar.LENGTH_SHORT)
//                .setAction(context.getString(R.string.item_swipe_undo), new View.OnClickListener() {
//                    @Override
//                    public void onClick(View v) {
//                        addItem(position, mItems.get(position));
//                    }
//                }).show();
    }

    public static class Holder extends RecyclerView.ViewHolder {
        @BindView(R.id.tv_title)
        TextView tvTitle;

        public Holder(@NonNull View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }
    }

}

File: client/src/main/java/com/example/client/_base/RvAdapter.kt
package com.example.client._base

import android.content.Intent
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.example.client.R
import kotlinx.android.synthetic.main.item_base.view.*

/**
 * Author: create by RhythmCoder
 * Date: 2020/7/10
 * Description:
 */
class RvAdapter(var list: List<Entry>) : RecyclerView.Adapter<RvAdapter.VH>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): VH {
        var view = LayoutInflater.from(parent.context).inflate(R.layout.item_base, parent, false)
        return VH(view)
    }

    override fun getItemCount(): Int {
        return list.size
    }

    override fun onBindViewHolder(holder: VH, position: Int) {
        holder.itemView.tv.text = list[position].title
        if (TextUtils.isEmpty(list[position].description)) {
            holder.itemView.description.visibility = View.GONE
        } else {
            holder.itemView.description.visibility = View.VISIBLE
            holder.itemView.description.text = list[position].description
        }
        holder.itemView.setOnClickListener {
            val intent = Intent(holder.itemView.tv.context, list[position].clazz)
            intent.putExtra("title", list[position].title)
            holder.itemView.tv.context.startActivity(intent)
        }
    }

    class VH(itemView: View) : RecyclerView.ViewHolder(itemView) {

    }

}